python/ray/tests/test_client_proxy.py

```diff
@@ -135,6 +135,8 @@ def test_correct_num_clients(call_ray_start):
     assert info._num_clients == 1
     run_string_as_driver(check_we_are_second.format(num_clients=2))
     ray.util.disconnect()
+    # Give server time to clean up old connection
+    time.sleep(5)
```

Comment:
Can we wait on a condition?

Reply:
Updated the cleanup logic so that the sleep isn't needed here.